src/pages/Admin/Room/Edit/RoomEditModal.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { useEffect, useState, useCallback, useMemo } from "react";
+import { Form, Modal, Button } from "react-bootstrap";
+import { useForm } from "react-hook-form";
+import ImageInput from "components/atoms/ImageInput";
+
+import { RoomTemplate, ROOM_TEMPLATES } from "settings";
+import { RoomEditModalProps } from "./RoomEditModal.types";
+
+import "./RoomEditModal.scss";
+
+export const RoomEditModal: React.FC<RoomEditModalProps> = ({
+  isVisible,
+  onClickOutsideHandler,
+  room,
+  submitHandler,
+  deleteHandler,
+}) => {
+  const [selectedTemplate, setSelectedTemplate] = useState<string>(
+    room.template!
```

Comment:
Using TypeScript's 'non-null assertion operator' is an unsafe antipattern in most places we use it currently. It's probably better to ban it from our codebase in general.

- https://dev.to/tmaximini/typescript-bang-operator-considered-harmful-3hhi

```suggestion
    room.template
```